distributed/scheduler.py

```diff
@@ -7485,7 +7495,17 @@ def decide_worker(
     if ts._actor:
         candidates = set(all_workers)
     else:
+        # Select all workers holding deps of this task
         candidates = {wws for dts in deps for wws in dts._who_has}
+        # Add up to 10 random workers into `candidates`, preferring idle ones.
+        worker_pool = valid_workers if valid_workers is not None else all_workers
+        if len(candidates) < len(worker_pool):
+            sample_from = idle_workers or worker_pool
```

Comment:
What if `valid_workers and idle_workers`?

Reply:
It seems like maybe we want to use `idle_workers or all_workers` above in line 7501